### PR TITLE
Use golangci-lint

### LINKED ISSUE #77: Use golangci-lint in favor of running individual tools
In [verify.yaml](https://github.com/shipwright-io/cli/blob/main/.github/workflows/verify.yaml), we today run a set of different tools. As decided in the community meeting from [2021/11/22](https://github.com/shipwright-io/community/issues/42#issuecomment-975581815), we should start to use [golangci-lint](https://golangci-lint.run/) and get rid of many of them.

Sibling issues:

- https://github.com/shipwright-io/build/issues/948
- https://github.com/shipwright-io/operator/issues/35

### PR DESCRIPTION
**Pull request by vancanhuit**
# Changes

Use [golangci-lint](https://github.com/golangci/golangci-lint-action) in CI.

Fixes #77

# Submitter Checklist

- [x] Includes tests if functionality changed/was added
- [x] Includes docs if changes are user-facing
- [x] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)  
- [x] Release notes block has been filled in, or marked NONE

# Release Notes
```release-note
NONE
```

### CODE REVIEW

**Review by SaschaSchwarze0 — CHANGES_REQUESTED** [review]
Hi @vancanhuit, thank you for engaging. Beside the issues that are now found, you will have to add a .golangci.yml to the project root. In there, you will have to enable gosec for example because this is disabled by default.

**Comment by vancanhuit** [comment]
Hi @SaschaSchwarze0, I will update accordingly.

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    runs-on: ubuntu-latest
+    env:
+      GO111MODULE: on
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v2
+      - name: Run Gosec Security Scanner
+        uses: securego/gosec@master
+        with:
+          args: ./...
```
Comment (SaschaSchwarze0) [inline .golangci.yml:20]: .golangci.yml is the [configuration file of golangci-lint](https://golangci-lint.run/usage/configuration/#config-file). You here pasted a GitHub workflow.

**Review by SaschaSchwarze0 — CHANGES_REQUESTED** [review]

File: .golangci.yml (hunk 1 above)
Comment (vancanhuit) [reply .golangci.yml]: Oh my bad. I pushed a new commit to enable `gosec`.

**Comment by vancanhuit** [comment]
Currently, the `Verify` step failed due to `errcheck`. Should we skip them for now to make it pass?

**Comment by otaviof** [comment]
> Currently, the `Verify` step failed due to `errcheck`. Should we skip them for now to make it pass?

I would say we make them pass 👍🏼  Thanks for tackling this, appreciated!

**Comment by vancanhuit** [comment]
I didn't touch any Go code but the unit test keeps failing. I also did run unit test locally and it passed.

**Comment by SaschaSchwarze0** [comment]
> I didn't touch any Go code but the unit test keeps failing. I also did run unit test locally and it passed.

The failure looks to be in the comparison of a map. I guess one [there](https://github.com/shipwright-io/cli/blob/main/pkg/shp/flags/map_value_test.go#L42-L43) has to stop stringifying and use [BeEquivalentTo](https://onsi.github.io/gomega/#beequivalenttoexpected-interface) to make it relyable.

**Comment by vancanhuit** [comment]
> > I didn't touch any Go code but the unit test keeps failing. I also did run unit test locally and it passed.
> 
> The failure looks to be in the comparison of a map. I guess one [there](https://github.com/shipwright-io/cli/blob/main/pkg/shp/flags/map_value_test.go#L42-L43) has to stop stringifying and use [BeEquivalentTo](https://onsi.github.io/gomega/#beequivalenttoexpected-interface) to make it relyable.

This is strange. I tried to fix it with new commit.

**Comment by vancanhuit** [comment]
@SaschaSchwarze0 CI now is green. Can we merge this PR?

**Review by SaschaSchwarze0 — APPROVED** [review]
/approve
/lgtm

**Pull request merged**